pages/integrations/output_framework.rst

```diff
@@ -0,0 +1,144 @@
+and then sent using the selected transport method.
+
+Messages are only passed to the Output Framework once they are done being processed in the 
+Graylog source cluster, at the same time the data is written to Elasticsearch.
+
+**On-Disk Journal**
+
+The Output Framework is equipped with a disk journal. This journal immediately persists 
+messages received from the Graylog Output system to disk before attempting to send them to
+the external receiver. This allows the Output Framework to keep receiving and reliably 
+queuing messages, even if the external receiver is temporarily unavailable due to
+network issues. The journal has many configuration options which will be described below.
+
+By default, journal data for Framework Outputs will be stored in the same directory used
+for the Input journal.  This directory is controlled by the ``message_journal_dir`` value
+in your Graylog configuration file.
+
+**Pipeline Integration**
+
+When creating or editing a Framework-based Output, you will have the option to select 
+a processing pipeline which will be executed on each message coming from the source 
+:doc:`stream <../streams>`. This pipeline can be used to filter out messages that you do
+not wish to forward.  It can also be used to add data to modify the contents of the outgoing
+message or to enrich it with additional data.
+
+**Outbound Payload Formatting**
+
+Prior to sending data out over the wire, Graylog must format the outgoing payload. Payload
+formatting options include:
+
+- **JSON**: The Output Framework will convert the message's key-value pairs into a JSON object.
+- **Pipeline-Generated**: The Output Framework will expect your pipeline to generate the 
+  outgoing payload and store it in the ``pipeline_output`` field of the message.  This can 
+  be accomplished in the pipeline by using the ``set_field`` :doc:`built-in function<../pipelines/functions>`.
+
+**Output Transports**
+
+- **Enterprise STDOUT**: Formatted messages will be displayed on the system's console. 
+  This is included primarily as a debugging tool for pipeline changes.
+- **Enterprise TCP Raw/Plaintext**: Formatted messages will be sent as UTF-8 encoded plain 
+  text to the configured TCP endpoint (IP address and port).
+- **Enterprise TCP Syslog**: Formatted messages will be sent as the ``MSG`` portion of a 
+  standard Syslog message per section 6.4 of the `Syslog specification <https://tools.ietf.org/html/rfc5424>`_.  
+  The Syslog message will be sent to the configured TCP endpoint (IP address and port).
+
+
+Output Configuration
+--------------------
+
+The Enterprise Output Framework is capable of processing messages at very high throughput 
+rates. Many hardware factors will affect throughput (such as CPU clock speed, number of 
+CPU cores, available memory, and network bandwidth). Several Output Framework configuration 
+options are available to help you tune performance for your throughput requirements and 
+environment.
+
```

Comment:
adding a list of the three different configuration types (with anchors) would be helpful to the user while navigating the docs